### ISSUE TITLE
aws-rds: support zero instances in DatabaseCluster

### ISSUE THREAD

**Issue by plumdog** [aws/aws-cdk#26724]
### Describe the feature

Both "old style" with `instances: number` and "new style" with `writer:` and `readers:` require that you end up with at least one instance in your cluster. Aurora doesn't actually require this, and further is actually a recommended approach to retaining data without spending money running instances. Per https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_DeleteCluster.html#USER_DeleteInstance.Empty:
> You can keep a cluster with no DB instances to preserve your data without incurring CPU charges for the cluster. You can quickly start using the cluster again by creating one or more new DB instances for the cluster. You can perform Aurora-specific administrative operations on the cluster while it doesn't have any associated DB instances. You just can't access the data or perform any operations that require connecting to a DB instance.

CDK should accommodate this.

### Use Case

Retaining Aurora data without running any instances, per https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_DeleteCluster.html#USER_DeleteInstance.Empty

### Proposed Solution

Adjust the props to accommodate this.

### Other Information

Workaround:
```typescript
const myDatabaseCluster = new rds.DatabaseCluster(this, 'MyDatabaseCluster', {
    // ...
    instanceProps: {
        instances: 1,
    },
});
myDatabaseCluster.node.tryRemoveChild('Instance1');
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.83.0

### Environment details (OS name and version, etc.)

n/a

**Comment by pahud** [comment]
Makes sense to me. Can you explain a little bit about this use case? I assume you would have created a cluster with at least 1 writer instance and you just want to re-deploy your CDK application to remove that writer instance but retain the cluster itself so when you create a new writer again the data would still be accessible. Is it correct?

Have you tried your workaround? What happened when you not specify the instance for the cluster? Will the  cluster be removed by cloudformation?

**Comment by plumdog** [comment]
@pahud My use case is to allow my application to have a "hibernate" mode. I have an application running on ECS and the Aurora database. I have a few different non-production configurations of this deployed separately that I want to be able to easily power up and down to accommodate testing.

The zero-instance-cluster approach is preferable because it removes the need to think about snapshots, which would require locating the right snapshot when powering up, and means my code can just have an on/off switch: I change the switch, push it to git, the pipeline deploys it, and everything comes back up. (The downside of this approach is that Aurora storage is 5x more expensive per GB than snapshots, $0.10 per GB-month for Aurora storage vs $0.021 per GB-month for snapshots (in us-east-1, at least). As I'm storing <10 GB across my several near-copies of this stack, this cost is negligible compared to the benefit of the pipeline being able to just switch the app back on.)

My workaround did work, and Cloudformation did not remove the cluster. This delete-the-cluster-when-deleting-the-last-instance behaviour appears to be unique to the web console, and this is documented here https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_DeleteCluster.html#USER_DeleteCluster.DeleteCluster.

**Comment by pahud** [comment]
Makes perfect sense to me and appreciate your workaround. This is currently marked as p2 which means we expect community to contribute the pull request. Feel free to submit any pull request for it. Thank you.
